### PR TITLE
[REST API] Added HTTP status codes for bad auth/requests

### PR DESCRIPTION
**Pull request by Sieberkev**
**Fixed issue:** https://github.com/jabbink/PokemonGoBot/issues/1139#issuecomment-241186713

**Changes made:**

* Added status codes on [bad requests 400](http://stackoverflow.com/questions/3290182/rest-http-status-codes-for-failed-validation-or-invalid-duplicate) (`the server cannot or will not process the request due to something that is perceived to be a client error`)/ auth 401
* Updated API.md file

### CODE REVIEW

**Comment by Sieberkev** [comment]
Pinging @JoeyyT to see if this has any negative consequences on your GUI (it shouldn't, with the codes it should be even be better to catch errors any errors (when the statuscode is not 200)).

File: API.md [hunk 1]
```diff
@@ -1,39 +1,58 @@
...
   - POST `/api/bot/{name}/start` => Start bot
 
-* Pokemon end-point :
+* Pokemon end-point:
   - GET `/api/bot/{name}/pokemons` => List all pokemons
   - POST `/api/bot/{name}/pokemon/{id}/transfer` => Transfer pokemon
-  - POST `/api/bot/{name}/pokemon/{id}/evolve` => Evolve pokemon
-  - POST `/api/bot/{name}/pokemon/{id}/powerup` => Power-up pokemon
+  - POST `/api/bot/{name}/pokemon/{id}/evolve` => Evolve pokemon (HTTP status code 400 (Bad Request) when not enough candy)
+  - POST `/api/bot/{name}/pokemon/{id}/powerup` => Power-up pokemon (HTTP status code 400 (Bad Request) when not enough candy)
```
Comment (FrancYescO) [inline API.md]: ..or stardust

**Comment by FrancYescO** [comment]
nothing will change for TTL5, rather will be better, cause it will show a failed message if not a 200 response received.

File: API.md (hunk 1 above)
Comment (Sieberkev) [inline API.md]: I've added a stardust check (that was missing).

File: src/main/kotlin/ink/abb/pogo/scraper/controllers/BotController.kt [hunk 2]
```diff
@@ -234,25 +242,31 @@ class BotController {
...
+            httpResponse: HttpServletResponse
+    ): String {
         val itemBag = service.getBotContext(name).api.inventories.itemBag
         val count = itemBag.items.find { it.itemId == ItemIdOuterClass.ItemId.ITEM_INCENSE_ORDINARY }?.count
+
         if (count == 0) {
+            httpResponse.status = HttpServletResponse.SC_BAD_REQUEST
             return "Not enough incense"
         } else {
             itemBag.useIncense()
```
Comment (FrancYescO) [inline src/main/kotlin/ink/abb/pogo/scraper/controllers/BotController.kt]: log if usage successfull? maybe also for luckyegg and location change request

**Pull request merged**